packages/analyticsdx-template-lint/src/schemas/readiness-schema.json

```diff
@@ -0,0 +1,347 @@
+                "type": ["null", "string"],
+                "description": "The image namespace.",
+                "defaultSnippets": [{ "label": "\"\"", "body": "${0}" }]
+              },
+              "name": {
+                "type": "string",
+                "description": "The image name."
+              }
+            },
+            "defaultSnippets": [
+              {
+                "label": "New backgroundImage",
+                "body": {
+                  "name": "${1:name}"
+                }
+              }
+            ]
+          },
+          "tags": {
+            "description": "Optional tags to associated with this requirement, which can be used to badge or group readiness results.",
+            "type": ["null", "array"],
+            "minItems": 0,
+            "items": {
+              "type": "string"
+            },
+            "defaultSnippets": [{ "label": "[]", "body": [] }]
+          }
+        },
+        "defaultSnippets": [
+          {
+            "label": "New templateRequirement",
+            "body": {
+              "expression": "${1}",
+              "failMessage": "${2}"
+            }
+          }
+        ]
+      },
+      "defaultSnippets": [{ "label": "[]", "body": [] }]
+    },
+    "definition": {
+      "description": "The readiness definition checks for this template. Each named entry here will be executed during the readiness check and available as ${Readiness.[name]} in the templateRequirements expression, the conditions in template-info, and rules.",
+      "type": ["null", "object"],
+      "additionalProperties": {
+        "description": "A readiness definition.",
+        "$ref": "#/definitions/definitionProps"
+      },
+      "defaultSnippets": [{ "label": "New definition", "body": { "${1}": { "type": "${2}" } } }]
+    }
+  },
+  "definitions": {
+    "definitionProps": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["type"],
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "The definition type.",
+          "enum": ["SobjectRowCount", "OrgDatasetRowCount", "AppDatasetRowCount", "OrgPreferenceCheck", "ApexCallout"],
+          "enumDescriptions": [
+            "A check against the number of objects in the org.",
+            "A check against the number of rows in a dataset in the org",
+            "A check against the number of rows in a dataset that will be created by this template in the app",
+            "",
+            "A check using a method on the template's apexCallback class."
+          ]
+        },
+        "filters": { "doNotSuggest": true },
+        "sobject": { "doNotSuggest": true },
+        "dataset": { "doNotSuggest": true },
+        "names": { "doNotSuggest": true },
+        "method": { "doNotSuggest": true },
+        "arguments": { "doNotSuggest": true }
+      },
+      "$comment": "Customize the type-specifc fields.",
+      "anyOf": [
```

Comment:
json schema doesn't really have discriminated union types, but we can kind of do it like this -- the same pattern is done for variables types and layout pages.